app/models/journeyDomain/locationOfGoods/LocationOfGoodsDomain.scala

```diff
@@ -50,153 +40,136 @@ object LocationOfGoodsDomain {
-        AddressPage.reader,
-        AddContactYesNoPage.filterOptionalDependent(identity)(UserAnswersReader[AdditionalContactDomain])
-      ).tupled.map((LocationOfGoodsZ.apply _).tupled)
-  }
-
-  case class LocationOfGoodsU(
-    typeOfLocation: LocationType,
-    unLocode: String,
-    override val additionalContact: Option[AdditionalContactDomain]
-  ) extends LocationOfGoodsDomain {
-
-    override val qualifierOfIdentification: LocationOfGoodsIdentification = UnlocodeIdentifier
-  }
-
-  object LocationOfGoodsU {
-
-    def userAnswersReader(typeOfLocation: LocationType): UserAnswersReader[LocationOfGoodsDomain] =
-      (
-        UserAnswersReader(typeOfLocation),
-        UnLocodePage.reader,
-        AddContactYesNoPage.filterOptionalDependent(identity)(UserAnswersReader[AdditionalContactDomain])
-      ).tupled.map((LocationOfGoodsU.apply _).tupled)
-  }
-
-  case class LocationOfGoodsT(
-    typeOfLocation: LocationType,
-    postalCodeAddress: PostalCodeAddress,
-    override val additionalContact: Option[AdditionalContactDomain]
-  ) extends LocationOfGoodsDomain {
-
-    override val qualifierOfIdentification: LocationOfGoodsIdentification = PostalCode
-  }
-
-  object LocationOfGoodsT {
-
-    def userAnswersReader(typeOfLocation: LocationType): UserAnswersReader[LocationOfGoodsDomain] =
-      (
-        UserAnswersReader(typeOfLocation),
-        PostalCodePage.reader,
-        AddContactYesNoPage.filterOptionalDependent(identity)(UserAnswersReader[AdditionalContactDomain])
-      ).tupled.map((LocationOfGoodsT.apply _).tupled)
-  }
+}
+
+case class LocationOfGoodsV(
+  typeOfLocation: LocationType,
+  customsOffice: CustomsOffice
+) extends LocationOfGoodsDomain {}
+
+object LocationOfGoodsV {
+
+  def userAnswersReader(typeOfLocation: LocationType): UserAnswersReader[LocationOfGoodsDomain] =
+    (
+      UserAnswersReader(typeOfLocation),
+      CustomsOfficeIdentifierPage.reader
+    ).tupled.map((LocationOfGoodsV.apply _).tupled)
+}
+
+case class LocationOfGoodsX(
+  typeOfLocation: LocationType,
+  identificationNumber: String,
+  additionalIdentifier: Option[String],
+  override val additionalContact: Option[AdditionalContactDomain]
+) extends LocationOfGoodsDomain {}
+
+object LocationOfGoodsX {
+
+  def userAnswersReader(typeOfLocation: LocationType): UserAnswersReader[LocationOfGoodsDomain] =
+    (
+      UserAnswersReader(typeOfLocation),
+      EoriPage.reader,
+      AddIdentifierYesNoPage.filterOptionalDependent(identity)(AdditionalIdentifierPage.reader),
+      AddContactYesNoPage.filterOptionalDependent(identity)(UserAnswersReader[AdditionalContactDomain])
+    ).tupled.map((LocationOfGoodsX.apply _).tupled)
+}
+
+case class LocationOfGoodsY(
+  authorisationNumber: String,
+  additionalIdentifier: Option[String],
+  override val additionalContact: Option[AdditionalContactDomain]
+) extends LocationOfGoodsDomain {
+
+  val typeOfLocation: LocationType = LocationType("Y", "Authorisation number")
```

Comment:
Can this change be reverted? Then pass the `typeOfLocation` into the userAnswersReader as before?